scripts/phased_training_resume.py

```diff
@@ -0,0 +1,202 @@
+    )
+    print(result.output)
+    if fail_on_phase1 or fail_on_phase2:
+        assert INTENTIONAL_TRAINING_FAILURE_MESSAGE in result.output
+        if fail_on_phase1:
+            assert_phase1(result)
+        elif fail_on_phase2:
+            assert_phase2_resumed(result)
+    elif fail_on_mt_bench:
+        assert INTENTIONAL_MT_BENCH_FAILURE_MESSAGE in result.output
+        assert_phase2_eval_resumed(result)
+    else:
+        assert_completion_resumed(result)
+    if fail_on_phase1 or fail_on_phase2 or fail_on_mt_bench:
+        assert result.exception is not None
+    assert result.exit_code == expected_exit_code
+
+
+def assert_phase1(result):
+    assert "Training Phase 1/2..." in result.output
+
+
+def assert_phase2(result):
+    assert "Training Phase 2/2..." in result.output
+
+
+def assert_phase2_eval(result):
+    assert "MT-Bench evaluation for Phase 2..." in result.output
+
+
+def assert_phase1_in_journal(result):
+    assert "SKIPPING: Training Phase 1/2; already in Journal" in result.output
+
+
+def assert_phase2_resumed(result):
+    assert_phase1(result)
+    assert_phase2(result)
+
+
+def assert_phase2_eval_resumed(result):
+    assert_phase2(result)
+    assert_phase2_eval(result)
+    assert_phase1_in_journal(result)
+
+
+def assert_completion_resumed(result):
+    assert_phase2_eval(result)
+    assert_phase1_in_journal(result)
+    assert "SKIPPING: Training Phase 2/2; already in Journal" in result.output
+    assert "Training finished! Best final checkpoint:" in result.output
+
+
+def test_phased_training_resume(
+    knowledge_data_path, skills_data_path, data_path, config
+):
+    phased_base_dir = os.path.join(data_path, "instructlab", "phased-resume")
+
+    print("Running training to fail on phase 1")
+    run_training_phase(
+        knowledge_data_path,
+        skills_data_path,
+        phased_base_dir,
+        config,
+        expected_exit_code=1,
+        fail_on_phase1=True,
+    )
+    print("Running training to fail on phase 2")
+    run_training_phase(
+        knowledge_data_path,
+        skills_data_path,
+        phased_base_dir,
+        config,
+        expected_exit_code=1,
+        fail_on_phase2=True,
+    )
+    print("Running training to fail on mt_bench")
+    run_training_phase(
+        knowledge_data_path,
+        skills_data_path,
+        phased_base_dir,
+        config,
+        expected_exit_code=1,
+        fail_on_mt_bench=True,
+    )
+    print("Running training to completion")
+    run_training_phase(knowledge_data_path, skills_data_path, phased_base_dir, config)
+
+    shutil.rmtree(phased_base_dir)
+
+    print("Finished phased training resume")
+
+
+if __name__ == "__main__":
+    parser = argparse.ArgumentParser(description="Phased Training")
+    parser.add_argument("--knowledge-data-path", help="Path to knowledge data path")
+    parser.add_argument("--skills-data-path", help="Path to skills data path")
+    parser.add_argument("--data-path", help="Data path")
```

Comment:
Which 'data' is this? If you're passing knowledge and skills already, what's this other one?

Reply:
Oh I see, it's the cache dir for phased_base_dir. Maybe changing the name to be more clear about it's usage would make sense.

Reply:
It's DATA_HOME in the e2e-ci script which comes from platformdirs.user_data_dir().

Would data_home be better?

Reply:
I went ahead and switched to data-home-path